### DISCUSSION TITLE
Made translations more flexible for specific setups

### DISCUSSION THREAD

**Discussion by prodigy7** [home-assistant/architecture#1156] [General]
Hey there!

I already wrote here https://github.com/danielperna84/hahomematic/discussions/1767 about me feature request, but SukramJ has kindly referred me here.

My concern: Currently the labels of the different week programs are hardcodes:

https://github.com/danielperna84/custom_homematic/blob/24d11c2e4f90e431208c7394a15a7267abe526ba/custom_components/homematicip_local/translations/de.json#L207-L212

It would be nice, if there would be a way, set a optional own custom label. Background in my case is, that I want to use week program "work" for weeks where I'm working outside, "holiday" if I'm at home. The current labels "WP 1", "WP 2" and so on are not really intuitive.

Is it possible to make labels configurable via HA and overwrite defaults?

**Comment by frenck** [comment]
👋  @prodigy7

> Is it possible to make labels configurable via HA and overwrite defaults?

Everything is possible, what is the architectural proposal you are raising on how are going to achieve this?

../Frenck

**Reply by prodigy7** [reply]
Sorry for the late response. 

So my thought is that it would be helpful if you could adapt certain texts to your own needs. I just had a quick look in the repository, but didn't see a translation module that you use. Anyway, I think a simple and transparent approach would be to integrate some kind of overlay. Example:

There would be a folder `translations-overlay` somewhere in the file system. In this folder you could then store a file `en.json`, for example, in which translations are entered that overwrite the actual core translations. If there is no translation to overwrite, the original is used. The advantage of this approach would be that you could automatically overwrite the modules with your own translations without having to adapt them.

Unfortunately, I do not currently know how exact translations work in HA and its modules. My current assumption is that the modules use existing functionalities.

What do you think?
